networkx/algorithms/swap.py

```diff
@@ -70,7 +70,9 @@ def directed_edge_swap(G, *, nswap=1, max_tries=100, seed=None):
     if nswap > max_tries:
         raise nx.NetworkXError("Number of swaps > number of tries allowed.")
     if len(G) < 4:
-        raise nx.NetworkXError("Graph has less than four nodes.")
+        raise nx.NetworkXError("DiGraph has less than four nodes.")
+    if len(G.edges) < 3:
+        raise nx.NetworkXError("DiGraph has less than 3 edges")
```

Comment:
Again, very minor grammatical nit: less -> fewer (might as well take the opportunity to clean up old grammar!)